src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -109,7 +114,7 @@ private void validateDatePartitionOptions(DatePartitionOptionsModel options,
                 columns.stream().filter(c -> targetColumn.equals(c.getName())).findFirst();
 
             if (matchingColumn.isPresent()) {
-                String colType = matchingColumn.get().getDatatype();
+                String colType = matchingColumn.get().getDatatype().toString();
 
                 if (!"DATE".equalsIgnoreCase(colType) && !"TIMESTAMP".equalsIgnoreCase(colType)) {
```

Comment:
Now that we have an enum, we can compare directly and not use `toString()`
```suggestion
                TableDataType colType = matchingColumn.get().getDatatype();

                if (colType != TableDataType.DATE  && colType != TableDataType.TIMESTAMP) {
```
This is also better in that we no longer have to rely on strings in the code that match an enum type elsewhere.